commit/src/pcs.rs

```diff
@@ -28,96 +31,60 @@ pub trait Pcs<Val: Field, In: MatrixRows<Val>> {
 
     type Error: Debug;
 
-    fn commit_batches(&self, polynomials: Vec<In>) -> (Self::Commitment, Self::ProverData);
-
-    fn commit_batch(&self, polynomials: In) -> (Self::Commitment, Self::ProverData) {
-        self.commit_batches(vec![polynomials])
-    }
-}
-
-pub type OpenedValues<F> = Vec<OpenedValuesForRound<F>>;
-pub type OpenedValuesForRound<F> = Vec<OpenedValuesForMatrix<F>>;
-pub type OpenedValuesForMatrix<F> = Vec<OpenedValuesForPoint<F>>;
-pub type OpenedValuesForPoint<F> = Vec<F>;
-
-pub trait UnivariatePcs<Val, EF, In, Challenger>: Pcs<Val, In>
-where
-    Val: Field,
-    EF: ExtensionField<Val>,
-    In: MatrixRows<Val>,
-    Challenger: FieldChallenger<Val>,
-{
-    fn open_multi_batches(
-        &self,
-        prover_data_and_points: &[(&Self::ProverData, &[Vec<EF>])],
-        challenger: &mut Challenger,
-    ) -> (OpenedValues<EF>, Self::Proof);
-
-    fn verify_multi_batches(
-        &self,
-        commits_and_points: &[(Self::Commitment, &[Vec<EF>])],
-        dims: &[Vec<Dimensions>],
-        values: OpenedValues<EF>,
-        proof: &Self::Proof,
-        challenger: &mut Challenger,
-    ) -> Result<(), Self::Error>;
-}
-
-/// A `UnivariatePcs` where the commitment process involves computing a low-degree extension (LDE)
-/// of each polynomial. These LDEs can be reused in other prover work.
-pub trait UnivariatePcsWithLde<Val, EF, In, Challenger>:
-    UnivariatePcs<Val, EF, In, Challenger>
-where
-    Val: Field,
-    EF: ExtensionField<Val>,
-    In: MatrixRows<Val>,
-    Challenger: FieldChallenger<Val>,
-{
-    type Lde<'a>: MatrixRows<Val> + MatrixGet<Val> + Sync
-    where
-        Self: 'a;
-
-    fn coset_shift(&self) -> Val;
-
-    fn log_blowup(&self) -> usize;
-
-    fn get_ldes<'a, 'b>(&'a self, prover_data: &'b Self::ProverData) -> Vec<Self::Lde<'b>>
-    where
-        'a: 'b;
+    fn natural_domain_for_degree(&self, degree: usize) -> Self::Domain;
```

Comment:
So this is like the "recommended" domain which might be most efficient, but generally any `Domain` should be supported by the impl?

Reply:
Yes